tracer/src/Datadog.Trace/IAST/Telemetry/IastInstrumentationMetricsHelper.cs

```diff
@@ -0,0 +1,75 @@
+// <copyright file="IastInstrumentationMetricsHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.Telemetry;
+using static Datadog.Trace.Telemetry.Metrics.MetricTags;
+
+namespace Datadog.Trace.Iast.Telemetry;
+
+internal static class IastInstrumentationMetricsHelper
+{
+    private static int _instrumentedPropagations = 0;
+    private static int _sinksCount = Enum.GetValues(typeof(IastInstrumentedSinks)).Length;
+    private static int _sourceTypesCount = Enum.GetValues(typeof(IastInstrumentedSources)).Length;
+    private static int[] _instrumentedSources = new int[_sourceTypesCount];
+    private static int[] _instrumentedSinks = new int[_sinksCount];
+    private static IastMetricsVerbosityLevel _verbosityLevel = Iast.Instance.Settings.IastTelemetryVerbosity;
+    private static bool _iastEnabled = Iast.Instance.Settings.Enabled;
+
+    public static void ReportMetrics()
+    {
+        if (_iastEnabled && _verbosityLevel != IastMetricsVerbosityLevel.Off)
```

Comment:
This scares me. The _only_ reason we can trust the source generator buffer array approach, is because we can guarantee the metrics are fixed and known ahead of time. As soon as you introduce the native side, that's not necessarily the case, because of version-mismatched scenarios.

Currently, in version-mismatch scenarios, there are two telemetry loops, one for each version. And if we change the IAST tags between versions... 💥 

I'm just not convinced we can (or necessarily should) have the communication between native and managed side for metrics. If anything, I think we should use libdatadog to send metrics and telemetry via the _native_ side only. That would remove all of these issues, plus we may need to do that at some point _anyway_. That could be a _much_ bigger job though 🙁